### PR TITLE
To Investigate: Debug flaky MFA announcement mailer tests

### PR DESCRIPTION
**Pull request by bettymakes**
### 🤷‍♀️ What problem are you trying to solve?
This PR is to follow up on the [MFA reminder mailer PR](https://github.com/rubygems/rubygems.org/pull/3166). @jchestershopify and I have been debugging this set of tests. When each one is run in isolation, they pass. However, when the whole suite of tests are run together, the tests start failing.

### 🔍 What we've observed so far:
It appears that the collection in `ActionMailer::Base.deliveries` is not always emptied correctly. We've attempted to clear the collection through calling `ActionMailer::Base.deliveries.clear`. Regardless of whether we clear the collection during `setup`, `teardown`, or inline within each test, it appears to not have made any difference. Some tests continue to fail.

We're opening this as a separate PR from the MFA reminder mailer since there's a time constraint to completing the mailer. If we end up shipping the mailer PR with only the one test which tests the expected path, we still want to backfill the other remaining tests.

### CODE REVIEW

**Comment by simi** [comment]
this works as intended https://github.com/rubygems/rubygems.org/commit/9406544c73cf8185290a5e9c1652cdef10249128 :pray: 

1. it doesn't use DelayedJob directly, but leaving send out to to ActiveJob
2. configures delayed job to be active job adapter
3. using active job helpers to perform background jobs
4. `capture_io` + `execute` on rake tasks to prevent reloading environment
5. manually including `ActionMailer::TestCase::ClearTestDeliveries`
6. loading rake tasks once in test helper

**Comment by bettymakes** [comment]
Ohh, this is great! Thanks for the guidance on this @simi 🙏. I'll take a closer look tomorrow morning to unpack this further and give it a go. 

Thanks again Josef!! 

(P.S. Once I get it all working, I'll go ahead and close this draft PR in favour of rolling these changes back into the [original mailer PR](https://github.com/rubygems/rubygems.org/pull/3166) where these tests originated from 😊.)

**Comment by bettymakes** [comment]
TL;DR - I was able to successfully resolve the flaky tests. So, I'm going ahead and closing this follow-up PR.

---

With @simi's guidance, it inspired me to dig further into my understanding of both Active Job and Action Mailer to ensure I fully understand how they work, how to set them up, and how to properly test them.

There was one discrepancy that stood out to me from Josef's approach vs. what's noted in the docs. In the [Action Mailer section](https://guides.rubyonrails.org/v6.1/active_job_basics.html#action-mailer) within the Active Job docs. Specifically, it notes that leveraging the async queue from a Rake task likely won’t work because the task will probably end before the "deliver-later"-triggered emails can be processed by ActiveJob. 

Since our mailers are being triggered through a Rake task, to avoid the potentially of the in-process thread pool being prematurely deleted, we can call `deliver` instead. Although, that doesn't actually delegate the email to be sent through ActiveJob 🤔. 

---

All that to say, I continued to debug and it appears implementing step 4 from [Josef's comment](https://github.com/rubygems/rubygems.org/pull/3167#issuecomment-1207111801) and also revising the test to be ActionMailer tests, the test suite began passing 🎉. 

With that said, I've gone ahead and updated the original PR with those two specific revisions, as well as the optimization of loading the rake tasks in the test helper.
